tests/framework/interface.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"testing"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+type testFramework interface {
+	TestMain(m *testing.M)
+	BeforeTest(testing.TB)
+	NewCluster(testing.TB) Cluster
+}
+
+type Cluster interface {
+	Close() error
+	Client() Client
```

Comment:
I think we can have two methods something like below,
```
ClusterClient() Client
MembeClient(memberIdx int) Client
```

Reply:
I didn't want to split those clients too early as there is no strong incentive. Forcing user to pick which member is correct unnecessary complicates the tests as it's not required by most test. I think that integration tests wrongly used member client and I hope that recent addition of cluster client should help in test simplification.